api/apps/api/src/modules/access-control/projects-acl/project-acl.service.ts

```diff
@@ -0,0 +1,82 @@
+import { InjectRepository } from '@nestjs/typeorm';
+import { Injectable } from '@nestjs/common';
+import { Repository } from 'typeorm';
+import { intersection } from 'lodash';
+
+import { UsersProjectsApiEntity } from '@marxan-api/modules/projects/control-level/users-projects.api.entity';
+import { Roles } from '@marxan-api/modules/users/role.api.entity';
+
+import { Permit, ProjectAccessControl } from './project-access-control';
+
+/**
+ * Debt: neither UsersProjectsApiEntity should belong to projects
+ * nor the Roles should belong to users
+ */
+@Injectable()
+export class ProjectAclService implements ProjectAccessControl {
+  private readonly canCreateScenarioRoles = [Roles.project_owner];
+  private readonly canEditScenarioRoles = [Roles.project_owner];
+  private readonly canViewSolutionRoles = [Roles.project_owner];
+  private readonly canPublishProjectRoles = [Roles.project_owner];
+  private readonly canViewProjectRoles = [
+    Roles.project_owner,
+    Roles.project_contributor,
+    Roles.project_viewer,
+  ];
+
+  private async getRolesWithinProject(
+    userId: string,
+    projectId: string,
+  ): Promise<Array<string>> {
+    const rolesToCheck = (
+      await this.roles.find({
+        where: {
+          projectId,
+          userId,
+        },
+        select: ['roleName'],
+      })
+    ).flatMap((role) => role.roleName);
+    return rolesToCheck;
+  }
+
+  constructor(
+    @InjectRepository(UsersProjectsApiEntity)
+    private readonly roles: Repository<UsersProjectsApiEntity>,
+  ) {}
+
+  // TODO: this will be changed in the following release of user requirements.
+  // For now, anyone should be able to create projects, regardless of having roles or not.
+  // In the future project creation will be limited to organization contributors, so this logic will be moved to the access control module
+  async canCreateProject(_userId: string, _projectId: string): Promise<Permit> {
+    return true;
```

Comment:
though structurally equivalent to `true` and `false` by definition, should we make use of `Allow` and `Deny` in these ACL functions, since we're using `Permit` in the function signature?

this would however imply a slightly more verbose syntax when we check the intersection of user roles with roles who can carry out an action.